io-engine/src/grpc/v1/nexus.rs

```diff
@@ -1152,33 +1155,42 @@ impl NexusRpc for NexusService {
     async fn list_rebuild_history(
         &self,
         request: Request<ListRebuildHistoryRequest>,
-    ) -> GrpcResult<RebuildHistoryMapResponse> {
+    ) -> GrpcResult<ListRebuildHistoryResponse> {
         let args = request.into_inner();
-        let count = args.count.unwrap_or_default() as usize;
         trace!("{:?}", args);
+        let count = args.count.unwrap_or(u32::MAX) as usize;
+        let end_time = args
+            .since_end_time
+            .map(chrono::DateTime::<chrono::Utc>::from);
         let rx = rpc_submit::<_, _, nexus::Error>(async move {
-            let history_map: HashMap<_, _> = nexus::nexus_iter()
+            let histories: HashMap<_, _> = nexus::nexus_iter()
                 .map(|nexus| {
-                    let uuid = nexus.uuid().to_string();
-                    let hist = nexus.rebuild_history();
+                    let hist = nexus.rebuild_history_guard();
                     let hist = hist
-                        .into_iter()
+                        .iter()
+                        .filter(|record| {
+                            end_time
+                                .map(|t| record.end_time > t)
```

Comment:
Do We need to have `>=` check ? When the query is usually from the user, it is usually considered to include the time input by the user. Though the chances of such instance is very rare :)